lib/Doctrine/DBAL/Driver/PDOConnection.php

```diff
@@ -92,17 +92,43 @@ public function quote(string $input) : string
     /**
      * {@inheritdoc}
      */
-    public function lastInsertId(?string $name = null) : string
+    public function lastInsertId() : string
     {
         try {
-            if ($name === null) {
-                return $this->connection->lastInsertId();
-            }
+            $lastInsertId = (string) $this->connection->lastInsertId();
+        } catch (\PDOException $exception) {
+            throw PDOException::fromNativePDOException($exception);
+        }
 
-            return $this->connection->lastInsertId($name);
+        if (! $this->isValidLastInsertIdOrSequenceNumber($lastInsertId)) {
+            throw new PDOException('No identity value was generated by the last statement.');
+        }
+
+        return $lastInsertId;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSequenceNumber(string $name) : string
+    {
+        try {
+            $sequenceNumber = (string) $this->connection->lastInsertId($name);
         } catch (\PDOException $exception) {
-            throw new PDOException($exception);
+            throw PDOException::fromNativePDOException($exception);
+        }
+
+        if (! $this->isValidLastInsertIdOrSequenceNumber($sequenceNumber)) {
```

Comment:
It's up to you really. If the PDO driver already has proper support for sequences on the given platform, I'm not sure whether there's a value in reinventing the wheel?

Reply:
Because unlike PDO, the concerns of driver and platform are separate in DBAL. If PDO exposed the functionality of generating SQL that we could use on the platform level, we would reuse it. Otherwise, we'll have to implement our own API ourselves.